sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -374,6 +374,9 @@ package object dsl {
         case plan => SubqueryAlias(alias, plan, None)
       }
 
+      def coalesce(num: Integer): LogicalPlan =
```

Comment:
They are used for the test cases in catalyst package, in which Dataset APIs are not available. Thus, that is why we add these DSL for test cases